crypto/rand/rand_win.c

```diff
@@ -43,51 +43,54 @@ int RAND_poll_ex(RAND_poll_fn cb, void *arg)
 {
 # ifndef USE_BCRYPTGENRANDOM
     HCRYPTPROV hProvider;
+    int ok = 0;
 # endif
-    DWORD w;
     BYTE buf[RANDOMNESS_NEEDED];
-    int ok = 0;
 
 # ifdef OPENSSL_RAND_SEED_RDTSC
     rand_read_tsc(cb, arg);
 # endif
 # ifdef OPENSSL_RAND_SEED_RDCPU
     if (rand_read_cpu(cb, arg))
-        ok++;
+        return 1;
 # endif
 
 # ifdef USE_BCRYPTGENRANDOM
     if (BCryptGenRandom(NULL, buf, (ULONG)sizeof(buf),
-                        BCRYPT_USE_SYSTEM_PREFERRED_RNG) != STATUS_SUCCESS)
-        return 0;
-    cb(arg, buf, sizeof(buf), sizeof(buf));
-    return 1;
+                        BCRYPT_USE_SYSTEM_PREFERRED_RNG) == STATUS_SUCCESS) {
+        cb(arg, buf, sizeof(buf), sizeof(buf));
+        return 1;
```

Comment:
No, @kaduk you are not misremembering.  More sources are considered better than less.  However, @richsalz is also correct in that once sufficient randomness has been gathered from _trusted_ sources, an early exit saves time and resources without impacting the quality.

In this instance, the user is declaring their trust for a source by enabling it.  Ideally, there would be a setting that enabled/disabled this short circuiting to pass the decision to the user, or better, a setting that allows a level of trust to be defined per source.  Possibly a PR for the future?  Another option might be to read less from each of several sources and repeat until the request is fulfilled.

The upside to the early exit is significant, reading /dev/random will block on many systems.  Especially in the presence of many competing attempts to read it.  This means a performance improvement and better sharing of a scarce system resource.

email is a better place I think.